account_multicurrency_revaluation/wizard/wizard_currency_revaluation.py

```diff
@@ -19,104 +19,82 @@
 
-class WizardCurrencyrevaluation(orm.TransientModel):
     _name = 'wizard.currency.revaluation'
 
-    _columns = {
-        'revaluation_date': fields.date(
-            'Revaluation Date',
-            required=True),
-        'journal_id': fields.many2one(
-            'account.journal',
-            'Journal',
-            domain="[('type','=','general')]",
-            help="You can set the default "
-                 "journal in company settings.",
-            required=True),
-        'currency_type': fields.many2one(
-            'res.currency.rate.type',
-            'Currency Type',
-            help="If no currency_type is selected,"
-            " only rates with no type will be browsed.",
-            required=False),
-        'label': fields.char(
-            'Entry description',
-            size=100,
-            help="This label will be inserted in entries description."
-            " You can use %(account)s, %(currency)s"
-                 " and %(rate)s keywords.",
-            required=True),
-    }
-
-    def _get_default_revaluation_date(self, cr, uid, context):
+    @api.model
+    def _get_default_revaluation_date(self):
         """
-        Get last date of previous period
+        Get last date of previous fiscalyear
         """
-        context = context or {}
-
-        period_obj = self.pool.get('account.period')
-        user_obj = self.pool.get('res.users')
-        cp = user_obj.browse(cr, uid, uid, context=context).company_id
-        # find previous period
-        current_date = date.today().strftime('%Y-%m-%d')
-        previous_period_ids = period_obj.search(
-            cr, uid,
+
+        fiscalyear_obj = self.pool.get('account.fiscalyear')
+        cp = self._uid.company_id
+        # find previous fiscalyear
+        current_date = fields.date.context_today
+        previous_fiscalyear = fiscalyear_obj.search(
             [('date_stop', '<', current_date),
-             ('company_id', '=', cp.id),
-             ('special', '=', False)],
+             ('company_id', '=', cp.id)],
             limit=1,
-            order='date_start DESC',
-            context=context)
-        if not previous_period_ids:
+            order='date_start DESC')
+        if not previous_fiscalyear:
             return current_date
-        last_period = period_obj.browse(
-            cr, uid, previous_period_ids[0], context=context)
-        return last_period.date_stop
+        return previous_fiscalyear.date_stop
 
-    def _get_default_journal_id(self, cr, uid, context):
+    @api.model
+    def _get_default_journal_id(self):
         """
         Get default journal if one is defined in company settings
         """
-        user_obj = self.pool.get('res.users')
-        cp = user_obj.browse(cr, uid, uid, context=context).company_id
-
-        journal = cp.default_currency_reval_journal_id
+        journal = self._uid.company_id.default_currency_reval_journal_id
         return journal and journal.id or False
```

Comment:
`return journal.id` should be enough as previous line should return you an empty record set or a record set with one record